lightning/src/routing/router.rs

```diff
@@ -1280,6 +1284,7 @@ impl<'a> PaymentPath<'a> {
 
 			let cur_hop = &mut self.hops.get_mut(i).unwrap().0;
 			cur_hop.next_hops_fee_msat = total_fee_paid_msat;
+			cur_hop.path_penalty_msat += extra_contribution_msat;
```

Comment:
It’s not higher cost with this solution because we don’t add the overpayment amount to the fees, just to the path contribution amount. The test fails without this as it will select an overpaying path when a path for the exact payment value is available.